src/coreclr/src/gc/env/volatile.h

```diff
@@ -74,7 +74,7 @@
 #if defined(__GNUC__)
 #if defined(_ARM_) || defined(_ARM64_)
 // This is functionally equivalent to the MemoryBarrier() macro used on ARM on Windows.
-#define VOLATILE_MEMORY_BARRIER() asm volatile ("dmb ish" : : : "memory")
```

Comment:
Hmm, actually, looking at the generated code for __sync_synchronize using compiler explorer, GCC 8.2 generates call here for arm (for arm64, it generates just dmb ish). That doesn't sound like a good thing perf-wise.
I'm not sure what does clang generate for ARM, I haven't checked yet.

Reply:
Sorry, I didn't understand what you mean by "GCC 8.2 generates call here ". Isn't this already generating a dmb instruction?

Reply:
ah, you are saying that instead of dmb ish; it generates a call to library function. It is probably because arm dmb support is complicated. I saw a userspace helper for this from the kernel.

Reply:
Right, that's what I've meant.

Reply:
I still wonder if using the __sync_synchronize is a good thing for GCC. Clang generates just "dmb ish" for this function, but as I've said, GCC generates a function call. The barriers are used usually at perf sensitive locations.

Reply:
Got it, let's abandon this. We don't have any performance data to say it is good or bad but I see what you are talking about.